### DISCUSSION TITLE
How can I add new backend endpoints?

### DISCUSSION THREAD

**Discussion by anh2111htdd** [supabase/supabase#1235] [Questions] [answered]
Based on my understanding, Supabase provides CRUD endpoints for entities in DB schema out of the box. In addition to that, I wish to add few more endpoints which have logic and query to Supabase DB e.g. integration with third-party services. Let me know if this is doable in Supabase. If it's not, is there any other service that I can use to host these extra endpoints while having access to Supabase? Thanks in advance.

**Comment by atnegrete** [comment, marked as answer]
You can use probably just about any hosting provider/cloud provider. You can instantiate the supabase js client on the server-side exactly the same way as you do on the client-side. You can use the secret key if you want to bypass row level security at your server side.

Most people on Supabase are familiar with Firebase also, if you are then you could setup some htttp firebase functions up there using the Supabase Client to aggregate your data + third party data.

Serverless functions are a highly requested feature in Supabase right now & still marked as coming soon 🎉 I hope that helps! :)

**Reply by anh2111htdd** [reply]
Yes, serverless function is what I was asking :) Glad that it will be coming soon. I'll try to instantiate the Supabase client on another backend for now!
